### PR TITLE
add scc template for jetstream job

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

add scc template for jetstream job

## Related Issues

https://github.com/astronomer/issues/issues/6457

## Testing

Refer issue ticket

## Merging

cherry-pick to release-0.35

### CODE REVIEW

**Comment by pgvishnuram** [comment]
@danielhoherd can you approve this PR

**Review by Simpcyclassy — APPROVED** [review]

File: tests/chart_tests/test_nats_jetstream.py [hunk 1]
```diff
@@ -182,3 +182,15 @@ def test_jetstream_hook_job_disabled(self, kube_version):
...
+        """Test that helm renders statefulset template for nats
+        jetstream with SA Disabled."""
+        docs = render_chart(
+            kube_version=kube_version,
+            values={},
+            show_only=[
+                "charts/nats/templates/jetstream-job-scc.yaml",
+            ],
+        )
+        assert len(docs) == 0
```
Comment (danielhoherd) [inline tests/chart_tests/test_nats_jetstream.py:196]: We should add a test with scc and jetstream enabled, and scc and stan enabled. We can probably just modify this test https://github.com/astronomer/astronomer/blob/98951f41ec/tests/chart_tests/test_nats_jetstream.py#L149-L169

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Code seems fine. Need to improve tests.

File: charts/nats/templates/jetstream-job-scc.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+###################################
+### Astronomer Jetstrem Job SCC ###
+###################################
+{{- if .Values.global.sccEnabled }}
+{{- if .Values.nats.createJetStreamJob -}}
+{{- if or .Values.nats.jetstream.enabled .Values.global.nats.jetStream.enabled .Values.global.stan.enabled }}
```
Comment (danielhoherd) [inline charts/nats/templates/jetstream-job-scc.yaml:6]: There are at least 6 possibilities of how these can be configured that have an impact on the code path. Currently there is only 1 test case. We should improve that.

**Comment by pgvishnuram** [comment]
we are good to merge

**Comment by danielhoherd** [comment]
Merged and cherry-picked to release-0.35

**Pull request merged**